osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+*
+*/
+
+#define _WIN32_DCOM
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+#include <Winternl.h>
+#pragma warning(push)
+/*
+C:\Program Files (x86)\Windows Kits\8.1\Include\um\DbgHelp.h(3190):
+warning C4091: 'typedef ': ignored on left of '' when no variable is declared
+*/
+#pragma warning(disable : 4091)
+#include <DbgHelp.h>
+#pragma warning(pop)
+#include <DbgEng.h>
+
+#include <boost/filesystem.hpp>
+
+#include <osquery/logger.h>
+#include <osquery/sql.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/windows/wmi.h"
+
+namespace fs = boost::filesystem;
+
+namespace osquery {
+namespace tables {
+
+const std::string kLocalDumpsRegKey =
+    "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\Windows Error "
+    "Reporting\\LocalDumps";
+const std::string kDumpFolderRegPath = kLocalDumpsRegKey + "\\DumpFolder";
+const std::string kFallbackFolder = "%TMP%";
+const std::string kDumpFileExtension = ".dmp";
+const unsigned long kNumStackFramesToLog = 10;
+const std::map<unsigned long long, std::string> kMinidumpTypeFlags = {
+    {0x00000000, "MiniDumpNormal"},
+    {0x00000001, "MiniDumpWithDataSegs"},
+    {0x00000002, "MiniDumpWithFullMemory"},
+    {0x00000004, "MiniDumpWithHandleData"},
+    {0x00000008, "MiniDumpFilterMemory"},
+    {0x00000010, "MiniDumpScanMemory"},
+    {0x00000020, "MiniDumpWithUnloadedModules"},
+    {0x00000040, "MiniDumpWithIndirectlyReferencedMemory"},
+    {0x00000080, "MiniDumpFilterModulePaths"},
+    {0x00000100, "MiniDumpWithProcessThreadData"},
+    {0x00000200, "MiniDumpWithPrivateReadWriteMemory"},
+    {0x00000400, "MiniDumpWithoutOptionalData"},
+    {0x00000800, "MiniDumpWithFullMemoryInfo"},
+    {0x00001000, "MiniDumpWithThreadInfo"},
+    {0x00002000, "MiniDumpWithCodeSegs"},
+    {0x00004000, "MiniDumpWithoutAuxiliaryState"},
+    {0x00008000, "MiniDumpWithFullAuxiliaryState"},
+    {0x00010000, "MiniDumpWithPrivateWriteCopyMemory"},
+    {0x00020000, "MiniDumpIgnoreInaccessibleMemory"},
+    {0x00040000, "MiniDumpWithTokenInformation"},
+    {0x00080000, "MiniDumpWithModuleHeaders"},
+    {0x00100000, "MiniDumpFilterTriage"},
+    {0x001fffff, "MiniDumpValidTypeFlags"}};
+const std::vector<MINIDUMP_STREAM_TYPE> kStreamTypes = {ExceptionStream,
+                                                        ModuleListStream,
+                                                        ThreadListStream,
+                                                        MemoryListStream,
+                                                        SystemInfoStream,
+                                                        MiscInfoStream};
+
+Status logAndStoreTID(const MINIDUMP_EXCEPTION_STREAM* stream,
+                      unsigned int& tid,
+                      Row& r) {
+  if ((stream == nullptr) || (stream->ThreadId == 0))
+    return Status(1);
+
+  r["tid"] = BIGINT(stream->ThreadId);
+  tid = stream->ThreadId;
+  return Status(0);
```

Comment:
nit: The default constructor for `Status` is an error code of `0`, so feel free to just use `Status()` throughout.